src/main/java/io/qdrant/client/QdrantClient.java

```diff
@@ -2380,6 +2388,65 @@ public ListenableFuture<DeleteSnapshotResponse> deleteFullSnapshotAsync(String s
 		addLogFailureCallback(future, "Delete full snapshot");
 		return future;
 	}
+	/**
+	 * Downloads a snapshot of a collection from the specified REST API URI and
+	 * saves it to the given
+	 * output path.
+	 *
+	 * @param outPath        The path where the snapshot will be saved.
+	 * @param collectionName The name of the collection.
+	 * @param snapshotName   The name of the snapshot. If null, the latest snapshot
+	 *                       will be downloaded.
+	 * @param restApiUri     The URI of the REST API. If null, the default URI
+	 *                       "http://localhost:6333"
+	 *                       will be used.
+	 */
+	public void downloadSnapshot(
+			Path outPath,
+			String collectionName,
+			@Nullable String snapshotName,
+			@Nullable String restApiUri) throws InterruptedException, IOException, ExecutionException {
+		String resolvedSnapshotName;
+		if (snapshotName != null) {
+			resolvedSnapshotName = snapshotName;
+		} else {
+			// Get the latest(0th) snapshot of the collection
+			List<SnapshotDescription> snapshots = listSnapshotAsync(collectionName).get();
+			if (snapshots.isEmpty()) {
+				throw new RuntimeException("No snapshots found");
+			}
+			resolvedSnapshotName = snapshots.get(0).getName();
+		}
+
+		String uri;
+		if (restApiUri != null) {
+			uri = String.format(
+					"%s/collections/%s/snapshots/%s", restApiUri, collectionName, resolvedSnapshotName);
+		} else {
+			uri = String.format(
+					"http://localhost:6333/collections/%s/snapshots/%s",
+					collectionName, resolvedSnapshotName);
```

Comment:
URI building should be done with correct encoding of path parts. If a `QdrantHttpClient` is implemented, the underlying HTTP client library used likely also has URI building methods.